### DISCUSSION TITLE
Can get.brier.survival be used on test (non-training) data for forest?

### DISCUSSION THREAD

**Discussion by KenleyData** [kogalur/randomForestSRC#136] [Q&A]
I would like to use the get.brier.survival function described in this vignette (https://luminwin.github.io/randomForestSRC/articles/survival.html). However, I would like to calculate the Brier score for my trained forest (created by the rsfrc function) applied to a test data set. Is this option available for get.brier.survival?

Thanks very much for any advice.

**Comment by ishwaran** [comment]
Yes, you can use the function `get.brier.survival` on a prediction object; i.e. a trained survival forest applied to test data:

```
data(veteran, package = "randomForestSRC")
train <- sample(1:nrow(veteran), round(nrow(veteran) * 0.80))
veteran.grow <- rfsrc(Surv(time, status) ~ ., veteran[train, ], ntree = 100) 
veteran.pred <- predict(veteran.grow, veteran[-train, ])
bs <- get.brier.survival(veteran.pred)
plot(bs$brier.score)
```

**Reply by KenleyData** [reply]
Thanks very much for your reply. I tried this on my data and it runs, but I think I'm confused with how to interpret the result. My understanding is that because the Brier score is a metric of accuracy, it should be undefinable past the final time of follow-up for the test data. In our case, we only have test data follow up up to 852 days. But when I run the recommended code (using "predict" for my test cohort, and then plugging the prediction into get.brier.survival) it produces a result with bs$brier.score$time giving a list of 150 numbers that range from 0 to 3960. I can't think of any special significance of time=3960. The maximum value of the "time" variable for the training cohort is 4053 and the maximum value of the "time" variable for the test cohort is 852.

In summary:

I was expecting:
bs$brier.score$time = c(1:852)

Instead I am getting:
bs$brier.score$time = [0 12 25 37 . . . . 3533 3648 3960]

I pasted my code below and attached a screenshot below of the brier.score in the R viewer. Thanks very much for any advice!

rsf_n <- rfsrc(formula_rf_n, data = early_cohort, ntree=500, mtry=20, nodesize=5, seed=1)
survival_predictions_n <- predict(rsf_n, newdata = late_cohort) 
bs <- get.brier.survival(survival_predictions_n)

![Screen Shot 2021-12-28 at 12 39 21 PM](https://user-images.githubusercontent.com/94707214/147596899-7e9a9298-f01f-4290-b354-b318d5db7cb2.png)

**Comment by ishwaran** [comment]
That's because the censoring distribution estimator needed for the calculation of the Brier score is based on the training data.  We do this since the test data could be very small or specialized and therefore a censoring estimator calculated from such test data would provide inaccurate estimates.   Also, as a general principle, the test data should not be used for forming the empirical metric used for calculating test performance.  Moreover, one always expects the training data to be the gold standard and so this is used for the construction.  Likewise, the Brier calculation for the test data are given for the original training data event times.  Thus to be clear, we use the event times of the deaths from the training data.  In general, all survival curves and survival estimates from RSF are always given based on these values.  This is the values stored in the object $time.interest.  Everything kicks off this master list.
